CurrentRange.c

```diff
@@ -0,0 +1,64 @@
+#include "CurrentRange.h"
+
+static int RangeValue[20][4];
+
+int cmpfunc (const void * value1, const void * value2) 
+{
+   return ( *(int*)value1 - *(int*)value2 );
+}
+
+int calcRangeCnt(int consecutivecnt,int rangecnt,int start,int end)
+{
+   if(consecutivecnt!=0)
+     {
+        rangecnt++;
+        printf("Range,Readings\n");
+        printf("%d - %d, %d",start,end,consecutivecnt+1);
+        RangeValue[rangecnt][0] = start;
+        RangeValue[rangecnt][1] = end;
+        RangeValue[rangecnt][2] = consecutivecnt+1;
+      }
+   return rangecnt;
+}
+
+int isConsecutive(int difference)
+{
+   int consecutive = 0;
+   if((difference==0) || (difference)==1)
+     consecutive=1;
+   return consecutive;
+}
+               
+   
+int * PrintRange(int array[],int arraysize)
+{ 
+   int rangecount=0;
+   int consecutivecount=0;
+   int start=array[0];
+   int end=array[0];
+   int index = 0;
+   int diff = 10;  
+  
+  qsort(array, arraysize, sizeof(int), cmpfunc);
+ 
+  for( index = 0 ; index < arraysize; index++ ) 
+     {   
+       diff = array[index+1] - array[index];        
```

Comment:
index+1 goes out of range whenever index points to the last element. 
This is a risky access since , the working of this never guaranteed. the value returned by array[index+1] is a garbage value and the failure due to this can be difficult to trace.